Avoid empty values for devices triggering events

jotakar opened the pull request:
In event scripts some devices has not a value for state (example thermostatMode). With thios modif if device is empty we use nValue for its state. So devicechanged["name-device"] has ever a right value to show.

jotakar committed:
Avoid empty values for devices triggering events

In event scripts some devices has not a value for state (example thermostatMode). With thios modif if device is empty we use nValue for its state. So devicechanged["name-device"] has ever a right value to show.

gizmocuz commented on main/EventSystem.cpp:
```diff
@@ -3670,6 +3672,14 @@ std::string CEventSystem::nValueToWording(const unsigned char dType, const unsig
+        	if (lstatus == "")
+        	{
+           		temp = nValue;
+           		ss << temp;
```
no need to use temp
please declase the stringstream here, no need to put it outside this scope

you can just do

ss << (int) temp

jotakar committed:
Update EventSystem.cpp

Remove useless variable temp and move ss declaration to where it's used.

gizmocuz commented on main/EventSystem.cpp:
```diff
@@ -3670,6 +3669,14 @@ std::string CEventSystem::nValueToWording(const unsigned char dType, const unsig
 		lstatus = sValue;
+		//OJO if lstatus  is still empty we use nValue for lstatus. ss for conversion
+		std::stringstream ss;
+        	lstatus = sValue;
```
It looks lstatus is assigned the value of sValue twice (as done above your comment)

jotakar committed:
Remove duplicate sentence

lstatus is assigned the same value twice.
Declaration of ss variable moved to inside the if block.

gizmocuz commented:
Thanks!

jotakar commented:
You’re welcome!

 

De: gizmocuz [mailto:notifications@github.com] 
Enviado el: martes, 03 de enero de 2017 8:10
Para: domoticz/domoticz
CC: jotakar; Author
Asunto: Re: [domoticz/domoticz] Avoid empty values for devices triggering events (#1092)

 

Thanks!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/domoticz/domoticz/pull/1092#issuecomment-270060699> , or mute the thread <https://github.com/notifications/unsubscribe-auth/ARrYdS3knk4dwJsuGxIGSeWawjJ5WT2Gks5rOfQ-gaJpZM4LYsmj> .Imagen quitada por el remitente.
